### ISSUE TITLE
SSL certificates aren't restored if Apache feature is disabled

### ISSUE THREAD

**Issue by iliaross** [virtualmin/virtualmin-gpl#857]
Hello Jamie.

Do we expect SSL certificates for the domain to be backed up and later restored on the destination system when a domain on the source system doesn't have the **Apache website enabled** feature turned on?

The source system had the **Apache website enabled** feature disabled, and it had SSL certificates used for Dovecot, Postfix, Webmin, and other services. Yet, after backing up the domain and restoring it on another system with Apache, the SSL certificates were not restored.

On the positive side, it didn't recreate those records in those services with missing files in them.

I think it's a bug. Even though we don't break anything, the restore process changes things, which isn't something a user might expect.

**Comment by jcameron** [comment]
Yes, this looks like a bug - I'll work on fixing it for a future release.

**Comment by iliajie** [comment]
I have applied all the latest patches, but it has started to fail now:

<img width="1130" alt="image" src="https://github.com/user-attachments/assets/a17d104d-4d4c-4481-bc0e-be491c45d435">

**Comment by jcameron** [comment]
Oops I forgot to create the SSL dirs - just checked in a fix for that.

**Comment by iliajie** [comment]
Thanks, Jamie! That seems to work now!

However, another issue has come up. The SSL certificates appear to be in place, but Virtualmin is still complaining and showing an extra button:

<img width="1101" alt="image" src="https://github.com/user-attachments/assets/4e69cb52-b2d9-4376-b65f-a225979fb984">
.. when clicked:
<img width="1111" alt="image" src="https://github.com/user-attachments/assets/4a237868-41ac-441d-a2c5-b4f01b20ff85">

Afterwards, the button disappeared when returned to SSL Certificates page.

It seems that `ssl.cert` was missing in the first place. Evidently another bug!

**Comment by jcameron** [comment]
So even after restoring the domain, was `ssl.cert` missing from that path?

If so, can you check of it was included in the backup file?

**Comment by iliajie** [comment]
> So even after restoring the domain, was `ssl.cert` missing from that path?

Yes, but I fixed it later.

The other problem I see is when a virtual server is created with the website feature disabled. It doesn't seem to create a self-signed SSL certificate at all.

For example:

<img width="1235" alt="image" src="https://github.com/user-attachments/assets/650276f4-e78d-4e08-a9df-6fbc237e45c7">

<img width="1204" alt="image" src="https://github.com/user-attachments/assets/3755c3d3-dece-4862-a1b8-e96b512edad2">

<img width="1235" alt="image" src="https://github.com/user-attachments/assets/fd1ea703-807c-480a-9f40-b30159b1fab6">

This issue happens despite SSL certificate sharing.

No SSL certificate files are created in `/etc/ssl/virtualmin` either.

**Comment by jcameron** [comment]
That's kind of expected - users can choose to create an SSL cert if needed, but the assumption is that if the domain has no website it's unlikely than an SSL cert will be needed.

**Comment by iliajie** [comment]
But the mail feature is enabled?

**Comment by jcameron** [comment]
I suppose we could add an option to always request / generate a cert..

**Comment by iliajie** [comment]
Yes, but in this case only if DNS is controlled by Virtualmin.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.
